src/main/java/org/opensearch/agent/tools/WebSearchTool.java

```diff
@@ -411,45 +430,54 @@ public String getType() {
+        if (isQueryEmpty) {
+            log.warn("Query is empty");
+            return false;
         }
-        boolean hasEndpoint = org.apache.commons.lang3.StringUtils.isEmpty(parameters.getOrDefault("endpoint", getDefaultEndpoint(engine)));
-        if (!hasEndpoint) {
-            throw new IllegalArgumentException("endpoint is empty");
+
+        boolean isEndpointEmpty = org.apache.commons.lang3.StringUtils.isEmpty(parameters.getOrDefault("endpoint", getDefaultEndpoint(engine)));
+        if (isEndpointEmpty) {
+            log.warn("Endpoint is empty");
+            return false;
         }
+
         if ("google".equalsIgnoreCase(engine)) {
             boolean hasEngineIdAndApiKey = parameters.containsKey("engine_id")
-                && !parameters.get("engine_id").isEmpty()
-                && parameters.containsKey("api_key")
-                && !parameters.get("api_key").isEmpty();
+                    && !parameters.get("engine_id").isEmpty()
+                    && parameters.containsKey("api_key")
+                    && !parameters.get("api_key").isEmpty();
             if (!hasEngineIdAndApiKey) {
-                throw new IllegalArgumentException("Google search engine_id or api_key is empty");
+                log.warn("Google search engine_id or api_key is empty");
+                return false;
             }
             return true;
         } else if ("duckduckgo".equalsIgnoreCase(engine)) {
             return true;
         } else if ("bing".equalsIgnoreCase(engine)) {
             boolean hasApiKey = org.apache.commons.lang3.StringUtils.isEmpty(parameters.get("api_key"));
             if (!hasApiKey) {
-                throw new IllegalArgumentException("Bing search api_key is empty");
+                log.warn("Bing search api_key is empty");
+                return false;
             }
             return true;
         } else if ("custom".equalsIgnoreCase(engine)) {
             String customApi = parameters.get("custom_api");
             String customResUrlJsonpath = parameters.get("custom_res_url_jsonpath");
             if (org.apache.commons.lang3.StringUtils.isEmpty(customApi)
-                || org.apache.commons.lang3.StringUtils.isEmpty(customResUrlJsonpath)) {
-                throw new IllegalArgumentException("custom search API is empty or result json path is empty");
+                    || org.apache.commons.lang3.StringUtils.isEmpty(customResUrlJsonpath)) {
+                log.warn("custom search API is empty or result json path is empty");
+                return false;
             }
+
             return true;
-        } else {
-            log.error("Unsupported search engine: {}", engine);
-            throw new IllegalArgumentException(String.format(Locale.ROOT, "Unsupported search engine: %s", engine));
         }
+
+        log.error("Unsupported search engine: {}", engine);
+        return false;
```

Comment:
This is not the correct place, but can you change the `create` method in Factory to add validation, it should basically similar to the validation here, e.g. for google, we need api_key and engine_id etc and also store these info in the tool.